### PR TITLE
Protect waveformmark.h's operator from causing segfaults

### PR DESCRIPTION
**Pull request by bencejuhaasz**
Every few hours while playing from a rekordbox usb, I got crashes because of this operator.
Checking for nullptr seems to fix the issue, I did a 5-6 hour test w/o any problem.
Not properly displaying hotcue tags seems to be a reasonable tradeoff to avoid a crash while playing live,
so I'm making a pull request. But if a simple 'false' is not the best return value, and we should check for the two marks separately, feel free to correct me.

### CODE REVIEW

**Comment by uklotzde** [comment]
Welcome and thank you for deciding to contribute.

The fix is relevant for release 2.3. Please create a separate, local bugfix branch based on `2.3` for your changes and then submit a new pull requests. Reusing `main` or any release branch directly for this purpose is not recommended. Unfortunately, GitHub PRs are tight to their branch names, that's why a new PR would be required then.

I will add a code comment to this PR.

**Comment by bencejuhaasz** [comment]
The crash doesn't happen in 2.3. It does in 2.4.

**Review by uklotzde — CHANGES_REQUESTED** [review]

**Comment by uklotzde** [comment]
We also need your permission before merging any code. Please sign our [Contributor Agreement](https://docs.google.com/forms/d/e/1FAIpQLScC9QG327sjLL0eWftmfGUasxFFLxg6LCXJ2xHDYRzFIRqyiw/viewform?formkey=dEpYN2NkVEFnWWQzbkFfM0ZYYUZ5X2c6MQ).

**Comment by uklotzde** [comment]
> The crash doesn't happen in 2.3. It does in 2.4.

Checking the pointers before dereferencing is still good practice and should also be added to 2.3. Just in case.

File: src/waveform/renderers/waveformmark.h [hunk 1]
```diff
@@ -114,20 +114,26 @@ class WaveformMark {
 typedef QSharedPointer<WaveformMark> WaveformMarkPointer;
 
 inline bool operator<(const WaveformMarkPointer& lhs, const WaveformMarkPointer& rhs) {
-    double leftPosition = lhs->getSamplePosition();
-    int leftHotcue = lhs->getHotCue();
-    double rightPosition = rhs->getSamplePosition();
-    int rightHotcue = rhs->getHotCue();
-    if (leftPosition == rightPosition) {
+    //A segfault in here crashes the whole program for me every few hours. Let's make sure we aren't comparing null pointers.
```
[the remark this answers is not in the document]
Comment (uklotzde) [reply src/waveform/renderers/waveformmark.h:117]: The debug assertions will help to reveal situations in which this bug occurs and only stop Mixxx during development, before any user notices.

**Comment by Swiftb0y** [comment]
IMO this is also just a workaround, knowing why these actually become `nullptr` would be much more interesting. Being able to catch that using `VERIFY_OR_DEBUG_ASSERT` would be a good first step towards that.

**Comment by bencejuhaasz** [comment]
The segmentation fault is still occuring, seems like this quick fix is not enough: 
```
Thread 1 "mixxx" received signal SIGSEGV, Segmentation fault.
__gnu_cxx::__ops::_Val_less_iter::operator()<QSharedPointer<WaveformMark>, QList<QSharedPointer<WaveformMark>>::iterator> (__it=...,   __val=<synthetic pointer>..., this=<synthetic pointer>) at /usr/include/c++/10/bits/predefined_ops.h:96
```

I don't know what to do with this, I'll probably just won't use waveform marks in my skin.

**Comment by Swiftb0y** [comment]
It would be great if you could run mixxx in a debugger to capture a backtrace for when these segfaults occur.
https://github.com/mixxxdj/mixxx/wiki/Creating%20Backtraces

**Comment by Swiftb0y** [comment]
I don't think the fault lies solely in the waveform marks, the culprit must be somewhere else. If you don't use any waveform marks, mixxx will probably just crash somewhere else.

**Comment by JoergAtGithub** [comment]
@bencejuhaasz This should be fixed by https://github.com/mixxxdj/mixxx/pull/11744 , can you test this please.
I close this PR in favor of https://github.com/mixxxdj/mixxx/pull/11744 therefore.

**Pull request closed without merging**